pkgs/applications/blockchains/whirlpool-gui/default.nix

```diff
@@ -0,0 +1,84 @@
+{ stdenv, fetchFromGitHub, callPackage, makeWrapper, makeDesktopItem
+, nodejs, yarn, fixup_yarn_lock, electron, jre, tor }:
+
+with stdenv.lib;
+
+let
+  yarnOfflineCache = (callPackage ./deps.nix {}).offline_cache;
+
+in stdenv.mkDerivation rec {
+	name = "whirlpool-gui";
+  version = "0.10.1";
+
+  src = fetchFromGitHub {
+    owner = "Samourai-Wallet";
+    repo = name;
+    rev = version;
+    sha256 = "ru6WJQRulhnQCPY2E0x9M6xXtFdj/pg2fu4HpQxhImU=";
+  };
+
+  nativeBuildInputs = [ makeWrapper fixup_yarn_lock nodejs yarn ];
+
+  configurePhase = ''
+    # Yarn and bundler wants a real home directory to write cache, config, etc to
+    export HOME=$NIX_BUILD_TOP/fake_home
+
+    # Make yarn install packages from our offline cache, not the registry
+    yarn config --offline set yarn-offline-mirror ${yarnOfflineCache}
+
+    # Fixup "resolved"-entries in yarn.lock to match our offline cache
+    fixup_yarn_lock yarn.lock
+
+    # install build dependencies and patch shebangs
+    yarn install --offline --frozen-lockfile --ignore-scripts --no-progress --non-interactive
+    patchShebangs node_modules/
+  '';
```

Comment:
Surely there are other yarn packages in Nix that have a well established pattern for this, and we don't need to re-implement it ourselves?

I'd take a look at `mkYarnPackage` and see if it can do what you need, like this one does:
https://github.com/NixOS/nixpkgs/blob/820cc39ad4dbadb75625b7623534ea09ba5536d8/pkgs/servers/monitoring/prometheus/default.nix#L13

Reply:
I have tried and it does not fit this package, as I just need to build statics and install only production dependencies